cmd/zapi/cmd/new/new.go

```diff
@@ -28,6 +30,7 @@ type Command struct {
 	*cmd.Command
 	kind     storage.Kind
 	datapath string
+	thresh   string
```

Comment:
Maybe make this a bytesValue.
```go
// bytesValue implements flag.Value.
type bytesValue int64

func (b *bytesValue) Set(s string) error {
	bytes, err := units.ParseStrictBytes(s)
	*b = bytesValue(bytes)
	return err
}

func (b bytesValue) String() string {
	return strconv.FormatInt(int64(b), 10)
}
```
You'll have to add it witth flag.(*FlagSet).Var instead of StringVar below, but it'll let you get rid of storageCfg and just say
```go
	req := api.SpacePostRequest{
		Name:     name,
		DataPath: c.datapath,
		Storage: &storage.Config{
			Kind: storage.Kind(c.kind),
			Archive: &storage.ArchiveConfig{
				CreateOptions: &storage.ArchiveCreateOptions{
					LogSizeThreshold: c.thresh.String(),
				},
			},
		},
	}
```